include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -380,15 +361,16 @@ struct __parallel_scan_submitter<_CustomName, __internal::__optional_kernel_name
         auto __final_event = __exec.queue().submit([&](sycl::handler& __cgh) {
             __cgh.depends_on(__submit_event);
```

Comment:
From my point of view, capture the source event by reference is not quite correct.
Here and in other places.
Better to capture by value.

Reply:
Do you mean the capture for the lambda being submitted here to capture `__submit_event`?
This is not new code, and is also how most submit calls in the library are done.  

If my understanding is correct, I think this is outside the scope of this PR if we want to make any change to that.

Reply:
ok.